File: src/stm_communication.cpp
/** stm_communication.cpp **/
#include <iostream>
#include "ros/ros.h"
#include <std_msgs/Int64.h>

using namespace std;

int main(int argc, char** argv){
    ros::init(argc, argv, "stm_communication");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<std_msgs::Int64>("counting", 1);

    std_msgs::Int64 hello;
    hello.data = 0;

    while(ros::ok()){
        pub.publish(hello);
        hello.data++;
        ros::Duration(0.5).sleep();
	    std::cout << "count=" << hello.data << std::endl;
    }

    ROS_INFO("== Terminate ! ==");
    return 0;
}

File: src/stm_communication.py
#!/usr/bin/env python3 
#coding:utf-8

import rospy
from std_msgs.msg import Int64

def publisher():
    rospy.init_node('talker')
    pub = rospy.Publisher('counting', Int64, queue_size = 10)
    count = 0
    while not rospy.is_shutdown():
            print("count=", count)
            pub.publish(count)
            rospy.sleep(0.5)
            count+=1
    
if __name__=='__main__':
    try:
        publisher()
    except rospy.ROSInterruptException:
        pass
